operators/hpe-csi-operator/destinations/community-operators/current-version/manifests/hpe-csi-operator.v2.4.1.clusterserviceversion.yaml

```diff
@@ -0,0 +1,732 @@
+apiVersion: operators.coreos.com/v1alpha1
+kind: ClusterServiceVersion
+metadata:
+  annotations:
+    alm-examples: |-
+      [
+        {
+          "apiVersion": "storage.hpe.com/v1",
+          "kind": "HPECSIDriver",
+          "metadata": {
+            "name": "hpecsidriver-sample"
+          },
+          "spec": {
+            "controller": {
+              "affinity": {},
+              "labels": {},
+              "nodeSelector": {},
+              "tolerations": []
+            },
+            "csp": {
+              "affinity": {},
+              "labels": {},
+              "nodeSelector": {},
+              "tolerations": []
+            },
+            "disable": {
+              "alletra6000": false,
+              "alletra9000": false,
+              "alletraStorageMP": false,
+              "nimble": false,
+              "primera": false
+            },
+            "disableNodeConfiguration": false,
+            "disableNodeConformance": false,
+            "disableNodeGetVolumeStats": false,
+            "imagePullPolicy": "IfNotPresent",
+            "iscsi": {
+              "chapPassword": "",
+              "chapUser": ""
+            },
+            "kubeletRootDir": "/var/lib/kubelet/",
+            "logLevel": "info",
+            "node": {
+              "affinity": {},
+              "labels": {},
+              "nodeSelector": {},
+              "tolerations": []
+            },
+            "registry": "quay.io"
+          }
+        }
+      ]
+    capabilities: Basic Install
+    categories: Storage
+    certified: "true"
+    createdAt: "2024-03-06T04:40:49Z"
+    description: A Container Storage Interface (CSI) driver for HPE storage platforms. The CSI driver allows you to use HPE storage with your preferred container orchestrator.
+    features.operators.openshift.io/cnf: "false"
+    features.operators.openshift.io/cni: "false"
+    features.operators.openshift.io/csi: "true"
+    features.operators.openshift.io/disconnected: "false"
+    features.operators.openshift.io/fips-compliant: "false"
+    features.operators.openshift.io/proxy-aware: "false"
+    features.operators.openshift.io/tls-profiles: "false"
+    features.operators.openshift.io/token-auth-aws: "false"
+    features.operators.openshift.io/token-auth-azure: "false"
+    features.operators.openshift.io/token-auth-gcp: "false"
+    olm.skipRange: '>=1.0.0 <2.4.1'
```

Comment:
Can we use this skip range such a way that customer can downgrade to previous version if possible ?  I think then customer will be able to see previous version in the drop down and will be able to downgrade to previous version if possible. Only for backward compatible version may be we can include this.

Reply:
The problem by allowing that is that we don't support lifecycle management of the "operator" and you'll end up with broken instances because of CRD changes etc. At least that that's what we saw in the past. It could work for Helm charts that don't have any parameter changes between releases and I can't remember when we had that last. This behavior might have improved over the years though.

Reply:
ok, yes sometimes it can lead to a broken instance since OLM does not support downgrade and there will not be any validation from OLM side.